CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ option(USE_OPENMP "Build with OpenMP support." ON)
 option(JVM_BINDINGS "Build JVM bindings" OFF)
 option(R_LIB "Build shared library for R package" OFF)
 ## Dev
+option(USE_DEBUG_VOMIT "Dump internal training results like gradients and predictions to stdout.
```

Comment:
@RAMitchell i wanted to limit its usage as much as possible, so no one can accidentality turn it on.  There's a limit of buffer size on most of terminals, it will flush your screen.   I can change to the wording, it's picked up from some other projects. ;-)